GrappleGirlGame/Character.lua

```diff
@@ -20,33 +21,58 @@ function Character:new(o, world, pos, speed)
     -- setmetatable(self.fixture:getUserData().val, self)
     -- self.fixture:getUserData().val = o
 
-
-    self.canJump = true;
+    self.canJump = true
 
     return o
 end
 
 function Character:draw()
-    x, y = self.body:getPosition()
-    love.graphics.circle("fill", x, y, 20)
+    local x, y = self.body:getPosition()
+    -- love.graphics.circle("fill", x, y, 20)
+    love.graphics.circle("fill", x, y, 10)
+    love.graphics.rectangle("fill", x - 10, y - 10, 20, 20)
 end
 
 function Character:update(dt)
-    f = 1000
+    local f = 1000
     if (love.keyboard.isDown("a")) then
         self.body:applyForce(-f, 0)
     end
     if (love.keyboard.isDown("d")) then
         self.body:applyForce(f, 0)
     end
 
-    if(self.canJump and love.keyboard.isDown("space"))then
+    if (self.canJump and love.keyboard.isDown("space")) then
         self.canJump = false
         self.body:applyLinearImpulse(0, -50)
     end
-
 end
 
-function Character:move(d)
+lastMouseState = {
+    b1 = false,
+    b2 = false,
+    b3 = false
+}
 
+
+grappleropejoint = {joint = nil, body=nil}
+function tmpHandleRope(dt)
+    if love.mouse.isDown(1) ~= lastMouseState.b1 then
+        lastMouseState.b1 = love.mouse.isDown(1)
+        if (love.mouse.isDown(1)) then
+            local pos = {}
+            local distance, x1, y1, x2, y2 = love.physics.getDistance(gGirl.fixture, grappleAnchor.fixture);
+            -- pos.b1x,pos.b1y = gGirl.body.getPosition();
```

Comment:
Try to avoid leaving commented-out lines of code here, and add comments indicating what is going on.

Reply:
rip missed that one when I went through and removed all the other ones

Reply:
wait no I didn't, you commented on an old piece of code

Reply:
My bad

Reply:
tisk tisk tisk